packages/terra-dropdown-button/src/_DropdownList.jsx

```diff
@@ -145,17 +149,28 @@ class DropdownList extends React.Component {
     return React.Children.map(this.props.children, (child, index) => React.cloneElement(child, {
       isActive: index === this.state.active,
       requestClose: this.props.requestClose,
+      ...(index === 0) && { 'data-terra-dropdown-first-list-item': true },
+      ...(index === React.Children.count(this.props.children) - 1) && { 'data-terra-dropdown-last-list-item': true },
```

Comment:
Something to consider. All dropdown list items could have the same data attribute instead of calculating a first and last item. In the scenario where a dropdown list contains only a single option both first and last data attributes would be applied.

```jsx
// Array index 0 is the first item and length - 1 is the last item. Assuming length is always >= 1. 
this.dropdownList.querySelectorAll('[data-terra-dropdown-list-item]'); 
```

Reply:
updated to queryselectorAll : [here](https://github.com/cerner/terra-core/blob/07833e70cea6c20f52e6cfb6b9bee03b4794740d/packages/terra-dropdown-button/src/DropdownButton.jsx#L112)